content/graph/reroot.cpp

```diff
@@ -1,51 +1,48 @@
-// output[r] = dp[r], where dp[v] :=
-//  fin(Sum_{child c of v, regarding root r} fromChild( dp[c] ))
+using W = ll; // edge weight type
+vector<vector<pair<int, W>>> adj;
 
-// To remove weights, remove every "w" and "W" and fix errors
 struct Reroot {
-  using D = /*todo*/; // dp value
-  using A = /*todo(often D)*/; //value from a vertex's child(ren)
-                               //(A,agg,e) commutative monoid
+	using T = ll; // dp type
 
-  A e = /*todo*/;
-  A fromChild(int v, int c, auto w, D dp_c) { /*todo*/ }
-  static A agg(A a, A b) { /*todo*/ }
-  D fin(int v, A chilsAgg) { /*todo*/ }
+	static constexpr T E = 0; // neutral element
```

Comment:
Does using `static constexpr` have an advantage (here)? (Now, T can't be string, vector<...> or set<...> anymore)

Reply:
Not really, you can change it if you want. Neutral element not being a constexpr is very rare though.

Reply:
```suggestion
	T E = 0; // neutral element
```